### PR TITLE
Feature/AddConvexHulls

### PR DESCRIPTION
**Pull request by mhuen**
Adds a parameter to specify the convex hull in the base module class. Also adds several convex hulls for default extensions around IceCube

### CODE REVIEW

**Comment by mjlarson** [comment]
Since this is an icetray plugin, I wonder whether it would be useful to use the version of this we already implement in `phys-services::ExtrudedPolygon` at https://github.com/icecube/icetray/blob/main/phys-services/public/phys-services/surfaces/ExtrudedPolygon.h. There are pybindings available at https://github.com/icecube/icetray/blob/main/phys-services/private/pybindings/Surface.cxx#L63-L69 and L94-99.

**Comment by mhuen** [comment]
This PR just adds some convenience around already existing implementations of the convex hull implementations in `geometry.py`. For the usage in the I3Modules, it may be worthwhile to see if this can use the `phys-services::ExtrudedPolygon`, or at least one should add the option to do so.  I'm not sure, however, if the `ExtrudedPolygon` is as general as the scipy implementation. In any case, this module is also used outside of IceTray environments.

File: ic3_labels/labels/base_module.py [hunk 1]
```diff
@@ -37,17 +47,23 @@ def Geometry(self, frame):
...
+                domPosDict[(31, 60)], domPosDict[(1, 60)],
+                domPosDict[(6, 60)], domPosDict[(50, 60)],
+                domPosDict[(74, 60)], domPosDict[(72, 60)],
+                domPosDict[(78, 60)], domPosDict[(75, 60)]
+                ]
+            self._convex_hull = ConvexHull(points)
+
+        elif isinstance(self._convex_hull, str):
+            self._convex_hull = getattr(detector, self._convex_hull)
+
```
Comment (pascalgutjahr) [inline ic3_labels/labels/base_module.py:73]: add assert, if convex hull is not correct

File: ic3_labels/labels/utils/detector.py [hunk 2]
```diff
@@ -1,103 +1,126 @@
...
+    [ 338.44000244,  463.72000122, 500],  # string 74
+    [ 101.04000092,  412.79000854, 500],  # string 72
+    [  22.11000061,  509.5       , 500],  # string 78
+    [-347.88000488,  451.51998901, 500],  # string 75
+    [-570.90002441, -125.13999939, 500],  # string 31
+]
+icecube_hull = ConvexHull(icecube_hull_points)
+
+# add extensions around IceCube
+for extension in [60, 150, 200, 300]:
```
Comment (pascalgutjahr) [inline ic3_labels/labels/utils/detector.py]: use optional key to define these extensions by the user

**Review by pascalgutjahr — COMMENTED** [review]
I suggest two minor fixes

**Comment by mhuen** [comment]
I created an issue to keep track of the requested feature to support `phys-services::ExtrudedPolygon` within IceTray and will merge this PR for now.

**Pull request merged**